rdrf/rdrf/urls.py

```diff
@@ -122,12 +122,12 @@
             name='login_assistance_complete'),
 
     re_path(r'^email_address/?$', EmailChangeRequestView.as_view(), name="email_address_change"),
+    re_path(r'^email_preferences/?$', EmailPreferencesView.as_view(), name='email_preferences'),
 
     # --- Token Authenticated views
-    re_path(r'^unsubscribe_all/(?P<username_b64>\w+)/(?P<token>[\w.:\-_=]+)/?$', UnsubscribeAllView.as_view(),
-            name='unsubscribe_all'),
-    re_path(r'^unsubscribe_preferences/(?P<username_b64>\w+)/(?P<token>[\w.:\-_=]+)/?$', EmailPreferencesView.as_view(), name='email_preferences'),
-    re_path(r'^email_preferences/?$', EmailPreferencesView.as_view(), name='email_preferences'),
+    re_path(r'^unsubscribe_all/(?P<username_b64>\w+)/(?P<token>[\w.:\-_=]+)/?$', UnsubscribeAllView.as_view(), name='unsubscribe_all'),
+    re_path(r'^unsubscribe_preferences/(?P<username_b64>\w+)/(?P<token>[\w.:\-_=]+)/?$', PublicEmailPreferencesView.as_view(), name='email_preferences'),
+
```

Comment:
@ScriptSmith , I was thinking we could simplify this by removing the token authenticated `email_preferences`, and just required the user to authenticate if they click the link from their email.
We'd keep the `unsubscribe_all` as a token authenticated view to retain our 'one click unsubscribe'. 

Or otherwise I'm still happy to proceed with this solution, but thought it was worth asking the question.

Reply:
Because it makes changing email preferences as friction-less as possible, I think it's worth it to keep the `PublicEmailPreferencesView`.  If there's some unnecessary complexity, we can address that without removing it.